PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/cosmosserverlessspark/spark/ui/livy/batch/KillCosmosServerlessSparkBatchJobAction.kt

```diff
@@ -31,27 +31,34 @@ import com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.Spar
 import com.microsoft.azuretools.ijidea.utility.AzureAnAction
 import com.microsoft.intellij.util.PluginUtil
 import org.apache.commons.lang3.exception.ExceptionUtils
+import rx.Observable
 
 class KillCosmosServerlessSparkBatchJobAction(private val account: AzureSparkServerlessAccount,
                                               private val job: SparkBatchJob) : AzureAnAction(AllIcons.Actions.Cancel), ILogger {
     override fun onActionPerformed(anActionEvent: AnActionEvent?) {
-        if (job.state() == SchedulerState.ENDED || job.state() == SchedulerState.FINALIZING) {
-            PluginUtil.displayInfoDialog("Kill Serverless Spark Job", "Can't kill spark job ${job.name()}. It's in '${job.schedulerState()}' state!")
-        } else {
-            account.killSparkBatchJobRequest(job.id()?.toString())
-                .subscribe(
-                    { resp ->
-                        if (resp.code >= 300) {
-                            PluginUtil.displayErrorDialog("Kill Serverless Spark Job", "Failed to kill spark job ${job.name()}. ${resp.message}")
-                        } else {
-                            PluginUtil.displayInfoDialog("Kill Serverless Spark Job", "Successfully killed Spark Job ${job.name()}!")
+        account.getSparkBatchJobRequest(job.id().toString())
+            .flatMap { respBatchJob ->
```

Comment:
`.doOnNext()` might be more convenient, which needn't to return self.

Reply:
> `.doOnNext()` might be more convenient, which needn't to return self.

But we created another observable in the else branch of the `flatMap` operator. If we use `doOnNext` instead, that observable instance need to be subscribed in the `doOnNext` operator, which is wierd.

Reply:
NP, since the kill method returns `Observable`